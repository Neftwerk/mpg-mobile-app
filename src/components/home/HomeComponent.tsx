import React from 'react';
import { ScrollView, Text, View } from 'react-native';

export const HomeComponent = () => {
	const [greeting, setGreeting] = React.useState('');
	const data = () => {
		fetch(`${process.env.EXPO_PUBLIC_API_URL}/hello`)
			.then((res) => {
				return res;
			})
			.then((res) => res.json())
			.then((res) => setGreeting(res.greeting))
			.catch((err) => console.log(err))
			.finally(() => console.log(greeting));
	};

	React.useEffect(() => {
		data();
	}, []);

	return (
		<ScrollView>
			<View testID="welcome-view">
				<Text className="p-5 font-bold">{greeting}</Text>
			</View>
		</ScrollView>
	);
};
