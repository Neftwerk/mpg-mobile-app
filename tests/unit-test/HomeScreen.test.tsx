import { render, waitFor } from '@testing-library/react-native';
import React from 'react';

import { HomeComponent } from '@/components/home/HomeComponent';

describe('HomeScreen', () => {
	test('Should home screen match snapshot', async () => {
		const tree = render(<HomeComponent />).toJSON();
		await waitFor(() => expect(tree).toMatchSnapshot());
	});

	test('Text renders correctly on HomeScreen', async () => {
		const { findByText } = render(<HomeComponent />);

		const greetingText = await findByText('Welcome!');

		expect(greetingText).toBeTruthy();
	});
});
