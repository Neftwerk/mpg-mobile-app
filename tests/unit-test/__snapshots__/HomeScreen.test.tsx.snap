// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Should home screen match snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      testID="welcome-view"
    >
      <Text
        className="p-5 font-bold"
      />
    </View>
  </View>
</RCTScrollView>
`;
